### ISSUE TITLE
spaCy underperforming after loading a custom model

### ISSUE THREAD

**Issue by GuillemGSubies** [explosion/spaCy#2854]
I am trying to load my custom embeddings into spaCy but its performance is worse than it should be. I am using a pretrained GloVe model: http://nlp.stanford.edu/data/glove.840B.300d.zip
## How to reproduce the behaviour

```python

from gensim.models import KeyedVectors
from gensim.scripts.glove2word2vec import glove2word2vec
import spacy
import numpy as np
from scipy import sparse
from sklearn.linear_model import LogisticRegression
from sklearn.datasets import fetch_20newsgroups
import string


def benchmark(name, clf=LogisticRegression(n_jobs=-1)):
    print('_' * 80)
    print("Training: ")
    print(clf)
    clf.fit(X_train, y_train)
    score = clf.score(X_test, y_test)
    print("score:   %0.3f" % score)
    print()
    return [name, score]

def mean_transform(X, model):
    transformed = None
    for elem in X:
        aux = ''.join(ch for ch in elem if ch not in set(string.punctuation))
        wordVectors = [np.array(model.get_vector(word)) for word in aux.split(' ') if word in model.vocab]
        if len(wordVectors) == 0:
            length = len(model[next(key for key in model.vocab)])
            wordVectors.append(np.zeros(length))
        meanVector = list(np.mean(wordVectors, axis=0))
        transformed = sparse.vstack([transformed, sparse.coo_matrix(meanVector)])
    return transformed

data_train = fetch_20newsgroups(subset='train', random_state=42)
data_test = fetch_20newsgroups(subset='test', random_state=42)
# split a training set and a test set
y_train, y_test = data_train.target, data_test.target

# After unizp
glove_input_file = 'glove.840B.300d.txt'
word2vec_output_file = 'glove.840B.300d.txt.word2vec'
glove2word2vec(glove_input_file0, word2vec_output_file0)
model_vanila = KeyedVectors.load_word2vec_format(word2vec_output_file, binary=False)
X_train = mean_transform(data_train.data, model_vanila)
X_test = mean_transform(data_test.data, model_vanila)
results_vanila = benchmark("Vanila")

#After: python -m spacy init-model en Custom --vectors-loc glove.840B.300d.txt.word2vec
model_custom = spacy.load('Custom')
X_train = [model_custom(elem).vector for elem in data_train.data]
X_test = [model_custom(elem).vector for elem in data_test.data]
results_spacy = benchmark("SpaCy custom")
```
When executing the above, the score will be higher for the non spaCy model.
## Info about spaCy

* **spaCy version:** 2.0.13
* **Platform:** Linux-4.4.0-130-generic-x86_64-with-debian-stretch-sid
* **Python version:** 3.6.6
* **Models:** en_vectors_web_lg, en
* Environment Information: Conda env with only the dependencies needed to execute above code

I have 2 guesses here:

* It can be related to #2281
* When I delete punctuation signs, the performance is better. 

If it were one of both guesses, it means that spaCy's algorithms can improve.

**Comment by honnibal** [comment]
What aspect of performance are you concerned about here? Speed, accuracy...? I can't tell what you expect to happen here, or indeed what happens when the code runs.

**Comment by GuillemGSubies** [comment]
Oh, sorry. I meant accuracy, however I do not know how much randomness can there be in this kind of models.

**Comment by honnibal** [comment]
I don't really get what's going on here, but it looks like spaCy shouldn't be doing much --- just associating each word with the vector from your pre-trained model. You might want to verify that the words are indeed mapping to the right vector.
